### PR TITLE
👌 Improve labels usage

### PR DESCRIPTION
**Pull request by deifyed**
## Description

- Fixes incorrect labels usage for Prometheus language. Previously we appended the label to the name. Now we use actual annotations
- Adds ability to use an indefinate amount of labels

## Motivation and Context

@yngvark wanted it

## How to prove the effect of this PR?

1. Run the server with `BASE_URL=http://localhost go run main.go`
2. Make a request to the service. For example like this:
    ```shell
    curl http://localhost:3000/v1/metrics/events \
	-X POST \
	-H "User-Agent: okctl" \
	-H "Content-Type: application/json" \
	-d '{"category": "commandexecution", "action": "scaffoldcluster", "labels":{"phase": "start"}}'
    ```
3. Check http://localhost:3000/z/prometheus and confirm that entries now are annotated with curly braces 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -14,5 +14,8 @@
 # Dependency directories (remove the comment below to include it)
```
[the remark this answers is not in the document]
Comment (deifyed) [reply .gitignore:14]: I've updated the label usage now. Please re-review

File: pkg/endpoints/metrics/metric_handler.go [hunk 2]
```diff
@@ -3,21 +3,22 @@ package metrics
...
 
-	"github.com/prometheus/client_golang/prometheus"
-
 	"github.com/gin-gonic/gin"
 	"github.com/oslokommune/okctl-metrics-service/pkg/core"
 )
 
-var counters = make(map[string]prometheus.Counter)
+func generateMetricHandler(cfg config.Config, logger *logrus.Logger) (gin.HandlerFunc, endpoints.TeardownFn) {
+	counters := NewMetricRegistry(cfg.LegalAgents)
```
Comment (yngvark) [inline pkg/endpoints/metrics/metric_handler.go:18]: I like the registry.

**Review by yngvark — APPROVED** [review]
Works like a charm :+1:

File: main.go [hunk 3]
```diff
@@ -41,7 +41,8 @@ func main() {
 
 	logger.Info("Server started")
 
-	router := sw.New(cfg, logger, specification)
+	router, teardownFn := sw.New(cfg, logger, specification)
```
[the remark this answers is not in the document]
Comment (deifyed) [reply main.go:44]: Totally agree. I have no idea why the linter doesn't pick it up. Maybe I'm missing a config file

**Pull request merged**